Tasks/AndroidSigningV2/Tests/L0AndroidSignAlignAndroidHomeNotSet.ts

```diff
@@ -1,31 +1,38 @@
-import ma = require('vsts-task-lib/mock-answer');
-import tmrm = require('vsts-task-lib/mock-run');
-import path = require('path');
+import * as path from 'path';
 
-let taskPath = path.join(__dirname, '..', 'androidsigning.js');
-let tr: tmrm.TaskMockRunner = new tmrm.TaskMockRunner(taskPath);
+import { TaskMockRunner } from 'vsts-task-lib/mock-run';
 
-tr.setInput('files', '/some/path/a.apk');
-tr.setInput('jarsign', 'false');
-tr.setInput('zipalign', 'true');
+import * as sinon from 'sinon';
 
-process.env['VSTS_TASKVARIABLE_KEYSTORE_FILE_PATH'] = '/some/store';
-process.env['ANDROID_HOME'] = '';
+const taskPath = path.join(__dirname, '..', 'androidsigning.js');
+const taskRunner = new TaskMockRunner(taskPath);
 
-// provide answers for task mock
-let a: ma.TaskLibAnswers = <ma.TaskLibAnswers>{
-    "findMatch": {
+const getInput = sinon.stub();
```

Comment:
taskRunner.setInput() seems better than this.

Reply:
`setInput` is what was causing [this issue](https://github.com/Microsoft/vsts-task-lib/issues/372).  I feel like it should just be mocked like everything else in task lib.  That removes another moving part in the tests.

Reply:
tr.setInput still reads better in the test code. Have you discussed options with @ericsciple on solutions for the linked issue.

Reply:
This was just to prove that we could mock inputs with Sinon.  I think the mocking can get moved into task-lib.  Then the test file would look like this:

```typescript
import * as path from 'path';

import { TaskMockRunner } from 'vsts-task-lib/mock-run';

const taskPath = path.join(__dirname, '..', 'androidsigning.js');
const taskRunner = new TaskMockRunner(taskPath);

taskRunner.setInputs({
    'files': '/some/path/a.apk',
    'keystoreFile': 'keystoreFileId',
    'keystorePass': 'pass1',
    'keystoreAlias': 'somealias',
    'keyPass': 'pass2',
    'jarsign': true,
    'zipalign': false
});

taskRunner.setAnswers('taskLib', {
    getVariable: {
        'KEYSTORE_FILE_PATH': '/some/store'
    },
    findMatch: {
        "/some/path/a.apk": [
            "/some/path/a.apk"
        ]
    },
    checkPath: {
        "/some/path/a.apk": true
    },
    which: {
        "jarsigner": "/usr/bin/jarsigner"
    },
    exec: {
        "/usr/bin/jarsigner -keystore /some/store -storepass pass1 -keypass pass2 -signedjar /some/path/a.apk /some/path/a.apk.unsigned somealias": {
            "code": 0,
            "stdout": "jarsigner output here"
        }
    }
});

taskRunner.run();
```

@ericsciple , what do you think?

Reply:
Cool, that helps visualize where we are going. I like this direction.

Reply:
Correct.  TaskLibAnswers taps into a set of prefabricated mocks on the task-lib side.  The new approach would replace TaskLibAnswers and the prefabricated mocks with Sinon on the task-lib side.